apps/emqx_rule_engine/src/emqx_rule_api_schema.erl

```diff
@@ -172,44 +172,58 @@ fields("node_metrics") ->
     [{"node", sc(binary(), #{desc => ?DESC("node_node"), example => "emqx@127.0.0.1"})}] ++
         fields("metrics");
 fields("ctx_pub") ->
+    Event = 'message.publish',
     [
-        {"event_type", event_type_sc(message_publish)},
+        {"event_type", event_type_sc(Event)},
+        {"event", event_sc(Event)},
```

Comment:
Not for this fix.  But it would improve the error message in case someone manually crafts the request and makes a mistake.

Reply:
> Not for this fix
This is because `event_type` has different shape from our hookpoints, and we can't change the input without breaking clients.  The new `event` field simply maps the API input to its matching hookpoint to be used when testing the rule.

Reply:
adding a hidden field would make it into cluster.hocon which might cause users to wonder what it is.

Reply:
this the HTTP API schema, not configuration schema.  does it affect `cluster.hocon` somehow?

Reply:
I just created a rule, and couldn't spot any traces of this in cluster.hocon.

Reply:
> this the HTTP API spec schema, not configuration schema. does it affect `cluster.hocon` somehow?

ok, sorry. did not notice it's for api.